src/components/Sidebars/Payback.js

```diff
@@ -242,7 +273,13 @@ function ProxyAndAllowanceCheck({
           isLoading={allowanceLoading}
           isComplete={hasAllowance}
           onToggle={() =>
-            setAllowance(maker, updateState, newTxListener, proxyAddress)
+            setAllowance(
+              maker,
+              updateState,
+              newTxListener,
+              proxyAddress,
+              setHasAllowance
```

Comment:
`setHasAllowance` was never passed and therefore never updated `hasAllowance` in `setAllowance()`